sql/mysqld.cc

```diff
@@ -6316,6 +6529,9 @@ void handle_connections_sockets()
     create_new_thread(thd);
     set_current_thd(0);
   }
+#ifdef HAVE_SYSTEMD
+  sd_notify(0, "STOPPING=1");
```

Comment:
btw, you didn't do sd_notify(WATCHDOG=1) anywhere.
Does it make sense to have? or it's useless for a DBMS?